src/trio/_path.py

```diff
@@ -32,219 +22,102 @@
-    _wrap_iter: list[str]
-    _forward: list[str]
-
-    def __init__(
-        cls, name: str, bases: tuple[type, ...], attrs: dict[str, object]
-    ) -> None:
-        super().__init__(name, bases, attrs)
-
-        cls._forward = []
-        type(cls).generate_forwards(cls, attrs)
-        type(cls).generate_wraps(cls, attrs)
-        type(cls).generate_magic(cls, attrs)
-        type(cls).generate_iter(cls, attrs)
-
-    def generate_forwards(cls, attrs: dict[str, object]) -> None:
-        # forward functions of _forwards
-        for attr_name, attr in cls._forwards.__dict__.items():
-            if attr_name.startswith("_") or attr_name in attrs:
-                continue
-
-            if isinstance(attr, (property, types.ModuleType)):
-                cls._forward.append(attr_name)
-            elif isinstance(attr, (types.FunctionType, types.BuiltinMethodType)):
-                wrapper = _forward_factory(cls, attr_name, attr)
-                setattr(cls, attr_name, wrapper)
-            else:
-                raise TypeError(attr_name, type(attr))
-
-    def generate_wraps(cls, attrs: dict[str, object]) -> None:
-        # generate wrappers for functions of _wraps
-        wrapper: classmethod | Callable[..., object]  # type: ignore[type-arg]
-        for attr_name, attr in cls._wraps.__dict__.items():
-            # .z. exclude cls._wrap_iter
-            if attr_name.startswith("_") or attr_name in attrs:
-                continue
-            if isinstance(attr, classmethod):
-                wrapper = classmethod_wrapper_factory(cls, attr_name)
-                setattr(cls, attr_name, wrapper)
-            elif isinstance(attr, types.FunctionType):
-                wrapper = thread_wrapper_factory(cls, attr_name)
-                assert isinstance(wrapper, types.FunctionType)
-                wrapper.__signature__ = inspect.signature(attr)
-                setattr(cls, attr_name, wrapper)
-            else:
-                raise TypeError(attr_name, type(attr))
-
-    def generate_magic(cls, attrs: dict[str, object]) -> None:
-        # generate wrappers for magic
-        for attr_name in cls._forward_magic:
-            attr = getattr(cls._forwards, attr_name)
-            wrapper = _forward_magic(cls, attr)
-            setattr(cls, attr_name, wrapper)
-
-    def generate_iter(cls, attrs: dict[str, object]) -> None:
-        # generate wrappers for methods that return iterators
-        wrapper: Callable[..., object]
-        for attr_name, attr in cls._wraps.__dict__.items():
-            if attr_name in cls._wrap_iter:
-                wrapper = iter_wrapper_factory(cls, attr_name)
-                assert isinstance(wrapper, types.FunctionType)
-                wrapper.__signature__ = inspect.signature(attr)
-                setattr(cls, attr_name, wrapper)
-
-
-@final
-class Path(metaclass=AsyncAutoWrapperType):
+class Path(pathlib.PurePath):
     """A :class:`pathlib.Path` wrapper that executes blocking methods in
```

Comment:
```suggestion
    """A :class:`pathlib.PurePath` wrapper that executes blocking methods in
```

I'm not too familiar with `pathlib`, does this have implications for back compat? Why can't this be `pathlib.Path`?

Reply:
It's not a `PurePath` wrapper though. It's a `PurePath` subclass *and* a `Path` wrapper.

Subclassing `Path` is impossible because the blocking methods (e.g. `absolute`) are not async, and replacing sync methods with async is forbidden by type checkers.

Now you mention it though, maybe we should stop documenting that it's a wrapper? That's an implementation detail. This is just an async `Path` implementation that slots into the `pathlib` class hierarchy as one would expect.

Reply:
I'm not too familiar with `pathlib` things (hence these slightly strange comments), but if you think we can, then "An async `pathlib.Path`" but more fleshed out might sound better.

Reply:
Check out my latest commit and comment, which makes it being a wrapper actually part of the public API, and enables explicit subclassing support to match what's coming in Python 3.13.

I can take or leave my last commit. On average, I think it's nice to support subclassing to match what the stdlib does, but I'm also cool to make it an implementation detail and tweak the docstring. I'll go with the majority here!